SasquatchMac/SasquatchMac/ViewControllers/TransmissionViewController.swift

```diff
@@ -0,0 +1,640 @@
+    child2Table.delegate = self
+    child2Table.dataSource = self
+    propertiesTable.delegate = self
+    commonTable.delegate = self
+    commonTable.dataSource = self
+    commonSelector.selectedSegment = 0
+    propertySelector.selectedSegment = 0
+
+    NotificationCenter.default.addObserver(self, selector: #selector(self.editingDidEnd), name: .NSControlTextDidEndEditing, object: nil)
+    let appName = Bundle.main.infoDictionary![kCFBundleNameKey as String] as! String
+
+    // Default target section.
+    let defaultTargetSection = TransmissionTargetSection()
+    defaultTargetSection.isDefault = true
+    defaultTargetSection.token = appName.contains("SasquatchMacSwift") ? kMSSwiftTargetToken : kMSObjCTargetToken
+
+    // Runtime target section.
+    let runtimeTargetSection = TransmissionTargetSection()
+    runtimeTargetSection.token = appName.contains("SasquatchMacSwift") ? kMSSwiftRuntimeTargetToken : kMSObjCRuntimeTargetToken
+
+    // Child 1.
+    let child1TargetSection = TransmissionTargetSection()
+    child1TargetSection.token = kMSTargetToken1
+
+    // Child 2.
+    let child2TargetSection = TransmissionTargetSection()
+    child2TargetSection.token = kMSTargetToken2
+
+    transmissionTargetSections = [defaultTargetSection, runtimeTargetSection, child1TargetSection, child2TargetSection]
+
+    // Common schema properties section
+    propertyValues = [String: [String]]()
+    collectDeviceIdStates = [String: Bool]()
+    let parentTargetToken = appName.contains("SasquatchMacSwift") ? kMSSwiftRuntimeTargetToken : kMSObjCRuntimeTargetToken
+    for token in [parentTargetToken, kMSTargetToken1, kMSTargetToken2] {
+      propertyValues[token] = Array(repeating: "", count: propertyKeys.count)
+      collectDeviceIdStates[token] = false
+    }
+    transmissionTargetMapping = [kMSTargetToken1, kMSTargetToken2, parentTargetToken]
+    commonSelector.target = self
+    commonSelector.action = #selector(onSegmentSelected)
+
+    // Target properties section
+    propertySelector.target = self
+    propertySelector.action = #selector(onSegmentSelected)
+  }
+
+  func numberOfRows(in tableView: NSTableView) -> Int {
+    if tableView == defaultTable {
+      return 3
+    }
+    else if tableView == commonTable {
+      return 5
+    }
+    return 4
+  }
+
+  func tableView(_ tableView: NSTableView, viewFor tableColumn: NSTableColumn?, row: Int) -> NSView? {
+    // Target properties section
+    if(tableView.tag == Section.TargetProperties.rawValue) {
+      guard let identifier = tableColumn?.identifier else {
+        return nil
+      }
+      let view = tableView.make(withIdentifier: identifier, owner: nil)
+      if (identifier == "value") {
+        let eventProperties = arrayController.content as! [EventProperty]
+        updateValue(property: eventProperties[row], cell: view as! NSTableCellView)
+      }
+      return view
+    }
+
+    tableView.headerView = nil
+
+    // Common schema properties section
```

Comment:
You can reverse these two `if`s for better readability, too

Reply:
@clpolet and I discussed this and it seems options are to either have crazy nested ifs or use `if !TransmissionTargets.shared.targetShouldSendAnalyticsEvents(targetToken: targetToken){` which I don't like because of the `!`, we favor leaving this as is.